### PR TITLE
Submit Documents Form and integration of apis

### PR DESCRIPTION
**Pull request by utpalBeehyv**
https://github.com/pucardotorg/dristi/issues/2636
https://github.com/pucardotorg/dristi/issues/2637
#2639
https://github.com/pucardotorg/dristi/issues/2640
https://github.com/pucardotorg/dristi/issues/2641

- [x] This PR has a proper title that briefly describes the work done
- [x] Please ensure the branch name follows naming convention - feature-githubissunumber-xxx, bug-githubissunumber-xxx, enhance-githubissunumber-xxx.
- [x] I have referenced the  github issues('s)
- [x] I performed a self-review of my own code
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added proper logs and comments for the developed code
- [ ] If this PR includes MDMS or workflow data changes:
  - [ ] I have added MDMS data changes to `support/<issue-number>-mdms.json`
  - [ ] I have added workflow data changes to `support/<issue-number>-workflow.json`

## Summary

- **Creation of New Routes for Submit Documents for Citizens**: Added new routes to handle the submission of documents by citizens within the application.
- **New Configuration for Submit Document**: Introduced a new configuration related to document submission, ensuring the process is streamlined and efficient.
- **Redirection and Form Disabling Based on Document Status**: Implemented functionality to redirect users or disable forms based on the status of the documents, such as if they are pending e-signature.
- Created Review Modal that will combine the documents submitted by user and displace it.
- Clicking on Sign button will create Evidence and it will move to Pending E-Sign state.
- After creating evidence, pushing artifact Number to params and then refetching the details. Creating a Pending E-Sign pending task for the user and clicking on it will redirect to review modal.
- Once evidence is created, user cannot update details.
- User can either upload document or Esign digitally. Then clicking on Finish button will update evidence to Submitted state and will show a success modal.
- Once Submitted, judge can mark and unmark a document as void with a reason.

## Preview
image
image
Screenshot from 2024-12-12 15-13-15
Screenshot from 2024-12-12 15-14-35
Screenshot from 2024-12-12 15-15-08
Screenshot from 2024-12-12 15-15-33
Screenshot from 2024-12-12 15-42-54
Screenshot from 2024-12-12 15-43-21

## Release Notes

- **New Features**
  - Introduced new components for document submissions, including `SubmissionDocuments`, `ReviewDocumentSubmissionModal`, `SubmissionDocumentEsign`, and `SubmissionDocumentSuccessModal`.
  - Added functionality for multi-file uploads and document review processes.
  - Enhanced case management with filing type integration across various components.
  - New role "EVIDENCE_EDITOR" added to user types, expanding permissions.

- **Bug Fixes**
  - Improved error handling and validation logic for document uploads and submissions.

- **Documentation**
  - Updated configuration files to reflect new roles and submission actions.

- **Chores**
  - Refactored existing components for better state management and user experience.

### CODE REVIEW

**Review by anirudh-0 — APPROVED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/components/SelectMultiUpload.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        alignItems: "center",
+        justifyContent: "center",
+        padding: "4px 8px",
+        cursor: "pointer",
+        fontFamily: "Arial, sans-serif",
+        fontSize: "14px",
+        color: "#007E7E",
+        height: "40px",
+        gap: "4px",
+        widht: "100%",
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/components/SelectMultiUpload.js:67]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/components/SelectMultiUpload.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        style={{
+          width: "16px",
+          height: "16px",
+          marginRight: "4px",
+          color: "#007E7E",
+        }}
+      />
+      <span style={{ color: "#007E7E", fontWeight: 600 }}>{t("UPLOAD_MORE")}</span>
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/components/SelectMultiUpload.js:80]: will move into scss file later

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/VoidSubmissionBody.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+const pStyle = {
+  fontFamily: "Roboto",
+  fontSize: "16px",
+  fontWeight: 400,
+  lineHeight: "24px",
+  textAlign: "left",
+  color: "#0A0A0A",
+  margin: "0px",
+};
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/VoidSubmissionBody.js:14]: will move into css file in later pr

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/utils/submissionDocumentsWorkflow.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export const SubmissionDocumentWorkflowAction = {
+  CREATE: "CREATE",
+  E_SIGN: "E-SIGN",
+};
+
+export const SubmissionDocumentWorkflowState = {
+  PENDING_ESIGN: "PENDING_E-SIGN",
+  SUBMITTED: "SUBMITTED",
+};
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/utils/submissionDocumentsWorkflow.js:9]: Status Coming from backend

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionDocuments.js [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+        {showSubmissionSuccessModal && (
+          <SubmissionDocumentSuccessModal
+            t={t}
+            handleClose={handleNextSubmission}
+            handleSuccessDownloadSubmission={handleSuccessDownloadSubmission}
+            documentSubmissionNumber={evidenceDetails?.artifactNumber}
+          />
+        )}
+      </div>
+      {showErrorToast && <Toast error={showErrorToast?.error} label={showErrorToast?.label} isDleteBtn={true} onClose={closeToast} />}
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionDocuments.js:436]: it's digit component

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionDocuments.js [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+    const isSignSuccess = localStorage.getItem("isSignSuccess");
+    // const applicationPDF = localStorage.getItem("applicationPDF");
+    if (isSignSuccess) {
+      setShowReviewModal(true);
+      // setShowsignatureModal(true);
+      // setApplicationPdfFileStoreId(applicationPDF);
+      localStorage.removeItem("esignProcess");
+      localStorage.removeItem("applicationPDF");
+    }
+  }, []);
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionDocuments.js:356]: we are saving esing fileStore and it's related in localStorage

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionDocuments.js [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+          isCompleted: true,
+          isAssignedRole: true,
+          assignedRole: ["EVIDENCE_CREATOR", "EVIDENCE_EDITOR"],
+        });
+        setShowReviewModal(false);
+        setShowSubmissionSuccessModal(true);
+      }
+    } catch (error) {
+      console.log("Error occured", error);
+      setShowErrorToast({ label: t("SOMETHING_WENT_WRONG"), error: true });
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionDocuments.js:291]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionDocuments.js [hunk 8]
```diff
@@ -0,0 +1,441 @@
...
+                uuid: userInfo?.uuid,
+                formdata,
+              },
+              workflow: {
+                action: SubmissionDocumentWorkflowAction.CREATE,
+              },
+            },
+          };
+          evidence = await DRISTIService.createEvidence(evidenceReqBody);
+          debugger;
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/pages/employee/SubmissionDocuments.js:246]: removed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/components/SubmissionDocumentEsign.js [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+        const uploadedFileId = await uploadDocuments(formData?.uploadSignature?.Signature, tenantId);
+        setSignedDocumentUploadID(uploadedFileId?.[0]?.fileStoreId);
+        setSignedId(uploadedFileId?.[0]?.fileStoreId);
+        setIsSigned(true);
+        setIsSignedHeading(true);
+      }
+    };
+
+    upload();
+  }, [formData]);
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/components/SubmissionDocumentEsign.js:77]: will check and done later

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/components/SubmissionDocumentEsign.js [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+    }
+  };
+
+  const cleanString = (input) => {
+    return input
+      .replace(/\b(null|undefined)\b/g, "")
+      .replace(/\s+/g, " ")
+      .trim();
+  };
+
```
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/submissions/src/components/SubmissionDocumentEsign.js:63]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/components/SelectMultiUpload.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (utpalBeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/components/SelectMultiUpload.js:67]: done

**Pull request merged**